server/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -122,8 +120,8 @@ public EqualityExtractor(EvaluatingNormalizer normalizer) {
             return null;
         }
 
-        List<Set<EqProxy>> comparisons = context.comparisonSet();
-        Set<List<EqProxy>> cp = Sets.cartesianProduct(comparisons);
```

Comment:
`Sets.cartesianProduct` guarantees to be in the "traditional", lexicographical order. Guava's Immutable Sets keep the creation order, Jdk Immutable Sets not always. Therefore I use a List now as result to keep the same behavior. This does not add any additional overhead. Guava's `CartesianSet` is internally backed up by a `CartesianList`.